lib/kino/process.ex

```diff
@@ -137,6 +139,205 @@ defmodule Kino.Process do
+
+        # Reject the rest
+        _ ->
+          false
+      end)
+      |> Enum.sort_by(fn {_type, timestamp, _from, _to, _message} ->
+        timestamp
+      end)
+
+    # The Tracer GenServer is no longer needed, shut it down
+    GenServer.stop(tracer_pid)
+
+    # Get all the participating actors in the trace along with their sequence diagram IDs
+    {participants_lookup, _idx} =
+      trace_events
+      |> Enum.reduce({%{}, 0}, fn {_type, _timestamp, from, to, _message}, acc ->
+        acc
+        |> maybe_add_participant(from)
+        |> maybe_add_participant(to)
+      end)
+
+    # Generate the Mermaid formatted list of participants
+    participants =
+      participants_lookup
+      |> Enum.map_join("\n", fn {pid, idx} ->
+        case Process.info(pid, :registered_name) do
+          nil ->
+            "participant #{idx} AS #35;PID#{:erlang.pid_to_list(pid)};"
+
+          {:registered_name, []} ->
+            "participant #{idx} AS #35;PID#{:erlang.pid_to_list(pid)};"
+
+          {:registered_name, name} ->
+            "participant #{idx} AS #{inspect(name)};"
+        end
+      end)
+
+    # Generate the mermaid formatted list of message events
+    messages =
+      trace_events
+      |> Enum.map_join("\n", fn {type, _timestamp, from, to, message} ->
+        normalize_message(type, from, to, message, participants_lookup)
+      end)
+
+    Markdown.new("""
+    ```mermaid
+    sequenceDiagram
+    #{participants}
+    #{messages}
+    ```
+    """)
+  end
+
+  defp maybe_add_participant({participants, idx}, pid) when is_pid(pid) do
+    if Map.has_key?(participants, pid) do
+      {participants, idx}
+    else
+      {Map.put(participants, pid, idx), idx + 1}
+    end
+  end
+
+  defp maybe_add_participant(acc, _) do
+    acc
+  end
+
+  defp maybe_activate(idx, {:spawn_request, _, _, _, _, _, _, _}), do: "\nactivate #{idx}"
+  defp maybe_activate(_, _), do: ""
+
+  defp maybe_deactivate(idx, {:EXIT, _, _}), do: "\ndeactivate #{idx}"
+  defp maybe_deactivate(_, _), do: ""
+
+  defp normalize_message(type, from, to, message, participants_lookup)
+       when is_map_key(participants_lookup, from) and is_map_key(participants_lookup, to) do
+    from_idx = participants_lookup[from]
+    to_idx = participants_lookup[to]
+
+    formatted_message = label_from_message(message)
+
+    case type do
+      :send ->
+        "#{from_idx}->>#{to_idx}: #{formatted_message}" <>
+          maybe_activate(to_idx, message) <>
+          maybe_deactivate(from_idx, message)
+
+      :receive ->
+        "#{to_idx}-->>#{from_idx}: #{formatted_message}"
+    end
+  end
+
+  defp normalize_message(_, _, _, _, _), do: ""
+
+  defp label_from_message(message) do
+    case message do
+      {:EXIT, _, reason} -> "EXIT: #{label_from_reason(reason)}"
+      {:spawn_request, _, _, _, _, _, _, _} -> "SPAWN"
+      {:down, _, reason} -> "DOWN: #{label_from_reason(reason)}"
```

Comment:
```suggestion
      {:DOWN, _, :process, _, reason} -> "DOWN: #{label_from_reason(reason)}"
```

The label that you were before was specific to Task.async_stream. :)